### PR TITLE
[Optimization][DeepSeekV3.2]Precompute the attention_mask_offset for Prefill in the Indexer

### PR DESCRIPTION
**Pull request by ShaneGZhu**
## Motivation
In DeepSeekV3.2's Indexer Prefill stage, the `attention_mask_offset` (i.e., the causal attention window `[ks, ke)` for each token) was previously computed via Python loops on CPU — iterating batch by batch and token by token. This becomes a bottleneck on the critical forward path when handling large batch sizes or long sequences.

This PR precomputes `attention_mask_offset` using a Triton kernel on GPU during `init_attention_metadata`, eliminating the CPU-side Python loop overhead and improving Prefill throughput for the Indexer attention backend.

> :bulb: If this PR is a Cherry Pick, the PR title needs to follow the format by adding the [Cherry-Pick] label at the very beginning and appending the original PR ID at the end. For example, [Cherry-Pick][CI] Add check trigger and logic(#5191)

> :bulb: 如若此PR是Cherry Pick，PR标题需遵循格式，在最开始加上[Cherry-Pick]标签，以及最后面加上原PR ID，例如[Cherry-Pick][CI] Add check trigger and logic(#5191)

## Modifications
- **`fastdeploy/model_executor/ops/triton_ops/indexer_update_attn_mask_offsets.py`** *(new file)*
  Implements the Triton kernel `update_indexer_attn_mask_offsets` that batch-computes the causal attention window `[ks, ke)` for all prefill tokens in a single GPU kernel launch. Also provides a Python reference implementation `ref_update_attn_mask_offsets` for correctness verification.
- **`fastdeploy/model_executor/layers/attention/dsa_attention_backend.py`**
  Adds a `_update_forward_meta` method that calls `update_indexer_attn_mask_offsets` to precompute `attention_mask_offset` and stores the result into `forward_meta.indexer_attn_mask_offsets`.
- **`fastdeploy/model_executor/forward_meta.py`**
  Adds `indexer_attn_mask_offsets` field to `ForwardMeta` to carry the precomputed offsets.

## Usage or Command
No API changes. The optimization is transparent to users. To verify correctness, run:
```
python -m pytest tests/model_executor/ops/triton_ops/test_indexer_update_attn_mask_offsets.py -v
```

## Accuracy Tests
The new Triton kernel is validated against the Python reference implementation `ref_update_attn_mask_offsets` in the unit tests, covering:
- Edge cases (single token, empty sequences)
- Mixed-batch scenarios with varying sequence lengths
No changes to model forward logic or kernel math — accuracy of model outputs is unaffected.

## Checklist

- [ ] Add at least a tag in the PR title.
  - Tag list: [`[FDConfig]`,`[APIServer]`,`[Engine]`, `[Scheduler]`, `[PD Disaggregation]`, `[Executor]`, `[Graph Optimization]`, `[Speculative Decoding]`, `[RL]`, `[Models]`, `[Quantization]`, `[Loader]`, `[OP]`, `[KVCache]`, `[DataProcessor]`, `[BugFix]`, `[Docs]`, `[CI]`, `[Optimization]`, `[Feature]`, `[Benchmark]`, `[Others]`, `[XPU]`, `[HPU]`, `[GCU]`, `[DCU]`, `[Iluvatar]`, `[Metax]`]
  - You can add new tags based on the PR content, but the semantics must be clear.
- [ ] Format your code, run `pre-commit` before commit.
- [ ] Add unit tests. Please write the reason in this PR if no unit tests.
- [ ] Provide accuracy results.
- [ ] If the current PR is submitting to the `release` branch, make sure the PR has been submitted to the `develop` branch, then cherry-pick it to the `release` branch with the `[Cherry-Pick]` PR tag.

### CODE REVIEW

File: fastdeploy/model_executor/ops/triton_ops/indexer_update_attn_mask_offsets.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import paddle
+import triton
+import triton.language as tl
+
+from fastdeploy.model_executor.ops.triton_ops.triton_utils import (
+    enable_compat_on_triton_kernel,
+)
+
+
+@enable_compat_on_triton_kernel
```
Comment (gongshaotian) [inline fastdeploy/model_executor/ops/triton_ops/indexer_update_attn_mask_offsets.py:26]: 1.不必要的注释清理掉
2. 改成英文的

Comment (ShaneGZhu) [reply fastdeploy/model_executor/ops/triton_ops/indexer_update_attn_mask_offsets.py:26]: ok

File: fastdeploy/model_executor/forward_meta.py [hunk 2]
```diff
@@ -164,6 +164,7 @@ class ForwardMeta:
 
     # for mla & dsa
     position_ids: Optional[paddle.Tensor] = None
+    indexer_attn_mask_offsets: Optional[paddle.Tensor] = None
     # for kvcache slot
     slot_mapping: Optional[paddle.Tensor] = None
```
Comment (gongshaotian) [inline fastdeploy/model_executor/forward_meta.py:169]: 1. indexer_attn_mask_offsets 是必须放在forward meta 里的吗？放attn meta data 里有啥问题
2. 这些注释都不规范，解释下语意
